src/java/org/apache/cassandra/index/sai/plan/Operation.java

```diff
@@ -302,14 +302,15 @@ FilterTree filterTree()
         @Override
         KeyRangeIterator rangeIterator(QueryController controller)
         {
-            KeyRangeIterator.Builder builder = controller.getIndexQueryResults(expressionMap.values());
-            for (Node child : children)
-            {
-                boolean canFilter = child.canFilter();
-                if (canFilter)
-                    builder.add(child.rangeIterator(controller));
-            }
-            return builder.build();
+            return controller.getIndexQueryResults(expressionMap.values());
+// FIXME
```

Comment:
is it needed?

Reply:
this is broken right now b/c getIndexQueryResults used to return an Intersection(Union), now it returns Union(Intersection)

but Caleb said it's not needed right now so I'll drop in the assert that he suggested

see https://the-asf.slack.com/archives/C018YGVCHMZ/p1684354319517459